tests/framework/e2e.go

```diff
@@ -78,15 +82,31 @@ func (e e2eRunner) NewCluster(ctx context.Context, t testing.TB, cfg config.Clus
 	if err != nil {
 		t.Fatalf("could not start etcd integrationCluster: %s", err)
 	}
-	return &e2eCluster{*epc}
+	return &e2eCluster{t, *epc}
 }
 
 type e2eCluster struct {
+	t testing.TB
 	e2e.EtcdProcessCluster
 }
 
-func (c *e2eCluster) Client() Client {
-	return e2eClient{e2e.NewEtcdctl(c.Cfg, c.EndpointsV3())}
+func (c *e2eCluster) Client(cfg clientv3.AuthConfig) (Client, error) {
+	etcdctl := e2e.NewEtcdctl(c.Cfg, c.EndpointsV3())
+	if !cfg.Empty() {
+		// use integration test client to validate if permissions are authorized
+		_, err := integration.NewClient(c.t, clientv3.Config{
+			Endpoints:   c.EndpointsV3(),
+			DialTimeout: 5 * time.Second,
+			DialOptions: []grpc.DialOption{grpc.WithBlock()},
+			Username:    cfg.Username,
+			Password:    cfg.Password,
+		})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Probably we do not need this. Reasons:
1. It seems not good for e2e test to depends on integration test;
2. The related test cases will get a denied response if the auth info isn't correct. So the pre-check isn't necessary.

Reply:
I agree it's not optimal that e2e test depending on integration test. This is an exception when auth config is not empty. 

Copying from past conversation between @serathius and me. 

From me
> Some existing auth tests checks unauthenticated errors. 

> Integration test will throw an error at client creation stage with `clientv3.New` https://github.com/etcd-io/etcd/blob/7ec336f80c9463fecd6d8ef15a9f05f3e8f6287a/client/v3/client.go#L439

> However, e2e test won't fail at client creation stage and fail at "got response" stage instead. 

> Also every request from e2e test will establish a new grpc connection to the server but integration test reuses existing one. In order to use a different user name and password, integration test needs to create another client. 

From Marek
> I think we can create `clientv3.Client` just for this.

ref. https://github.com/etcd-io/etcd/pull/14041#discussion_r874629440

Does this makes sense to you?

Reply:
> It seems not good for e2e test to depends on integration test;

It's not tests, it's e2e tests using integration client, which is just a wrapper for clientv3 library. There is nothing wrong with e2e tests using client library for some things that are not surfaced in etcdctl. To complete migration of tests to new framework this is even required as Some tests cannot be expressed using just e2e or integration tools

> I don't think e2e and integration test must follow the same pattern or behavior

But it would make testing and reasoning about them much simpler.